src/parser.rs

```diff
@@ -1,35 +1,63 @@
 use html5ever::{self, Attribute};
 use html5ever::tree_builder::{TreeSink, NodeOrText, QuirksMode};
 use std::borrow::Cow;
+use std::fs::File;
+use std::path::Path;
+use std::io::Read;
 use string_cache::QualName;
 use typed_arena::Arena;
 
+
 use tree::Node;
+use std::option;
+
+pub struct Html<F = IgnoreParseErrors> where F: FnMut(Cow<'static, str>) {
+    opts: ParseOpts<F>,
+    data: option::IntoIter<String>,
+}
 
+impl Html  {
+    pub fn from_string(string: &str) -> Html {
```

Comment:
This should be `pub fn from_string<S: Into<String>>(string: S) -> Html {` so that if the user has an owned `String` they can give it directly without copying.